gcloud/monitoring/timeseries.py

```diff
@@ -0,0 +1,706 @@
+
+# Features intentionally omitted from this first version of the client library:
+#   - Creating time series.
+#   - Natural representation of distribution values.
+
+import collections
+import copy
+import datetime
+import itertools
+
+import six
+
+from gcloud.monitoring.metric import Metric
+from gcloud.monitoring.resource import Resource
+
+TOP_RESOURCE_LABELS = [
+    'project_id',
+    'aws_account',
+    'location',
+    'region',
+    'zone',
+]
+
+
+class Query(object):
+    """Query object for listing time series.
+
+    The preferred way to construct a query object is using the
+    :meth:`~gcloud.monitoring.client.Client.query` method
+    of the :class:`~gcloud.monitoring.client.Client` class.
+
+    :type client: :class:`gcloud.monitoring.client.Client`
+    :param client: The client to use.
+
+    :type metric_type: string
+    :param metric_type: The metric type name. The default value is
+        ``"compute.googleapis.com/instance/cpu/utilization"``, but
+        please note that this default value is provided only for
+        demonstration purposes and is subject to change.
+
+    :type resource_type: string
+    :param resource_type: An optional resource type to filter by.
+        For example: ``"gce_instance"``
+
+    :type end_time: :class:`datetime.datetime` or string or None
+    :param end_time: The end time (inclusive) of the time interval
+        for which results should be returned, as a datetime object.
+        The default is the start of the current minute. If the
+        ``days``/``hours``/``minutes`` parameters are not used, the
+        end time can alternatively be provided as a timestamp string
+        in RFC3339 UTC "Zulu" format.
+
+    :type start_time: :class:`datetime.datetime` or string or None
+    :param start_time: An optional start time (exclusive) of the time
+        interval for which results should be returned, as either a
+        datetime object or a timestamp string. If omitted and no
+        non-zero duration is specified, the interval is a point in
+        time. If any of ``days``, ``hours``, or ``minutes`` is non-zero,
+        these are combined and subtracted from the end time to determine
+        the start time.
+
+    :type days: integer
+    :param days: The number of days in the time interval.
+
+    :type hours: integer
+    :param hours: The number of hours in the time interval.
+
+    :type minutes: integer
+    :param minutes: The number of minutes in the time interval.
+    """
+
+    DEFAULT_METRIC_TYPE = 'compute.googleapis.com/instance/cpu/utilization'
+
+    def __init__(self, client,
+                 metric_type=DEFAULT_METRIC_TYPE, resource_type=None,
+                 end_time=None, start_time=None,
+                 days=0, hours=0, minutes=0):
+        if end_time is None:
+            end_time = datetime.datetime.utcnow().replace(second=0,
+                                                          microsecond=0)
+
+        if days or hours or minutes:
```

Comment:
The important use case is

    query = client.query(minutes=5)

Asking the user to construct a `timedelta` somehow would spoil this.

For programmatic use, given `end_time` and `duration` (a `datetime` and a `timedelta`), it's easy enough to do this:

    query = client.query(end_time=end_time, start_time=end_time - duration)